### DISCUSSION TITLE
Sample management

### DISCUSSION THREAD

**Discussion by tripsylon** [bangcorrupt/hacktribe#313] [Q&A] [answered]
Hello there, Is there a way to see which samples (under what number In the all sample file) are used in a pattern, or how many times is a sample used in the all sample file? I have about 160 track live set prepared and about 300 samples in the machine and I use maybe only 60 of them. I want to erase all the unused ones and replace them, but going through all the patterns manually would take me like 10 hours. Thank you for all the tips.

**Comment by bangcorrupt** [comment, marked as answer]
I think [e2_recode_sample_pat.py](https://github.com/bangcorrupt/hacktribe/blob/main/scripts/e2_recode_sample_pat.py) by @untergeekDE does something like this, but I haven't used it.  It may need some modification for your purpose.

**Reply by untergeekDE** [reply]
Are you using a run-of-the-mill E2S, or the Hacktribe already? If you have just converted to Hacktribe, and still have your original .e2sallpat and e2sSample.all files pre-conversion, you might use these two scripts:

```python e2_sample_all_to_wav.py -s e2sSample.all```
(Breaks up the monolithic sample file into lots of individual .WAV files you may reimport in case you need them)
```python e2_recode_sample_pat.py -s e2sSample.all -p my_patterns.e2sallpat -m```
(Generates a minimal copy of the sample library with all unused samples removed)

The functions are all in the code; would be quite easy to whip up a Colab which you could use to analyze and rework your files. I won't promise but give me a couple of days.

**Comment by tripsylon** [comment]
Hello, first of all thank you for the replies and I am sorry for the late response, I was quite busy at the time and forgotten to reply…
 I’m running Hacktribe on the red sampler. Like I said I just need to find out which samples are used and which are just taking up memory. 
I do realize this is easier said than done. But am thinking that maybe this kind of data chart could be a byproduct of some other operations🙃. 
Please take in mind that I am a total noob when it goes to programming… 
Also I use a Mac for computer work.
Thank you 🙏!
